chips/nrf5x/src/ble_advertising_driver.rs

```diff
@@ -6,19 +6,15 @@
 //! The advertisement interval is configured from the user application.
 //! The allowed range is between 20 ms and 10240 ms, lower or higher values will
 //! be set to these values. Advertisements are sent on channels 37, 38 and 39
-//! which are currently controlled by the chip.
+//! which are controlled by this driver. the chip just notifies the capsules via two
+//! interfaces: RxClient and TxClient for events
+//! .
 //!
-//! The total size of the combined payload is 31 bytes, the capsule ignores payloads
-//! which exceed this limit. To clear the payload, the `ble_adv_clear_data`
-//! function can be used. This function clears the payload, including the name.
-//!
-//! Only start and send are asynchronous and need to use the busy flag.
-//! However, the synchronous calls such as set tx power, advertisement interval
-//! and set payload can only by performed once the radio is not active.
-//! The reason why is that they can be interleaved by an interrupt
+//! The total size of the combined payload is 31 bytes, the driver ignores payloads
+//! which exceed this limit.
 //!
 //! ### Allow system call
-//! Each advertisement type corresponds to an allow number from 0 to 0xFF which
+//! Each advertesement type corresponds to an allow number from 0 to 0xFF which
 //! is handled by a giant pattern matching in this module
```

Comment:
This giant pattern match needs to be documented, so that someone writing a userland driver can reference it.